os/drivers/compression/compress.c

```diff
@@ -0,0 +1,140 @@
+#include <tinyara/config.h>
+
+#include <tinyara/fs/fs.h>
+#include <tinyara/fs/ioctl.h>
+#include <tinyara/compression.h>
+
+#include <errno.h>
+#include <debug.h>
+#include <assert.h>
+
+/****************************************************************************
+ * Function Prototypes
+ ****************************************************************************/
+
+static int comp_ioctl(FAR struct file *filep, int cmd, unsigned long arg);
+static ssize_t comp_read(FAR struct file *filep, FAR char *buffer, size_t len);
+static ssize_t comp_write(FAR struct file *filep, FAR const char *buffer, size_t len);
+
+/****************************************************************************
+ * Data
+ ****************************************************************************/
+static const struct file_operations compress_fops = {
+	0,                          /* open */
+	0,                          /* close */
+	comp_read,                  /* read */
+	comp_write,                 /* write */
+	0,                          /* seek */
+	comp_ioctl              /* ioctl */
+#ifndef CONFIG_DISABLE_POLL
+	, 0                         /* poll */
+#endif
+};
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+static ssize_t comp_read(FAR struct file *filep, FAR char *buffer, size_t len)
+{
+	return 0;
+}
+
+static ssize_t comp_write(FAR struct file *filep, FAR const char *buffer, size_t len)
+{
+	return 0;
+}
+
+
+/************************************************************************************
+ * Name: comp_ioctl
+ *
+ * Command description:
+ * 	 COMPIOC_COMPRESS - for compression of input data
+ * 	 COMPIOC_DECOMPRESS - for decompression of input data
+ * 	 COMPIOC_GET_COMP_TYPE - for getting compression type
+ * 	 COMPIOC_GET_COMP_NAME - for getting compression type in string
+ * 
+ * Arguments:
+ * 	 filep is ioctl fd, cmd is required command, arg is required argument for
+ * 	 the command. For compression and decompression, arg is a struct compress_header
+ * 	 pointer in which user stores input_buffer, input_size data.
+ * 	 For compression and decompression, It is required to allocate output_buffer and 
+ * 	 assign output_size of struct compress_header argument before passing to the function.
+ * 	 For compression, output size should be equal to input size.  
+ * 	 For getting compression_type, arg is NULL.
+ * 	 For getting compression_name, arg is a char pointer that needs to be 
+ * 	 allocated memory of size COMP_NAME_SIZE (compression.h).
+ *
+ * Description:
+ * 	 The ioctl method for compress apis. For compression and decompression,
+ * 	 ouput result is stored in outpu_buffer and output_size of arg(struct compress_header *).
+ * 	 For compression type, it just returns the compression type value.
+ * 	 For compression name, it stores the string in the char pointer that has been
+ * 	 cast to arg.
```

Comment:
Let us make readme document to tell the following. 
- In case of compression what should be the output buffer size.
- In case of decompression, what how to find output buffer required size?
- If output size is not known, then how much buffer must be allocated?
- If allocated buffer is smaller than required size, then how it will work? How can user perform decompression of complete data?
- What is the return value and value of in-out parameters for each case?